src/main/scala/com/mozilla/telemetry/streaming/CrashesToInflux.scala

```diff
@@ -0,0 +1,192 @@
+        )
+
+        val influxTags = ListMap(
+          "submissionDate" -> metadata.submissionDate,
+          "appVersion" -> metadata.appVersion,
+          "appName" -> metadata.appName,
+          "displayVersion" -> ping.getDisplayVersion.getOrElse(""),
+          "channel" -> metadata.normalizedChannel,
+          "country" -> metadata.geoCountry,
+          "osName" -> metadata.os.getOrElse(""),
+          "osVersion" -> ping.getOsVersion.getOrElse(""),
+          "architecture" -> ping.getArchitecture.getOrElse("")
+        )
+
+        val outputString = measurementName +
+          influxTags.map { case (k, v) => s"$k=$v" }.mkString(",", ",", "") +
+          influxFields.map { case (k, v) => s"$k=$v" }.mkString(" ", ",", "") +
+          " " +
+          timestamp
+
+        Array[String](outputString)
+      }
+    }
+  }
+
+  def getParsedPings(pings: DataFrame, raiseOnError: Boolean, measurementName: String,
+                     channels: List[String] = defaultChannels,
+                     appNames: List[String] = defaultAppNames): Dataset[String] = {
+    import pings.sparkSession.implicits._
+
+    pings.flatMap( v => {
+      try {
+        parsePing(Message.parseFrom(v.get(0).asInstanceOf[Array[Byte]]), channels, appNames, measurementName)
+      } catch {
+        case _: Throwable if !raiseOnError => Array[String]()
+      }
+    }).as[String]
+  }
+
+  def sendStreamCrashes(spark: SparkSession, opts: Opts): Unit = {
+    val httpSink = new RawHttpSink(opts.url(), Map())
+
+    val pings = spark
+      .readStream
+      .format("kafka")
+      .option("kafka.bootstrap.servers", opts.kafkaBroker())
+      .option("failOnDataLoss", opts.failOnDataLoss())
+      .option("kafka.max.partition.fetch.bytes", 8 * 1024 * 1024) // 8MB
+      .option("spark.streaming.kafka.consumer.cache.maxCapacity", kafkaCacheMaxCapacity)
+      .option("subscribe", TelemetryKafkaTopic)
+      .option("startingOffsets", opts.startingOffsets())
+      .load()
+
+    getParsedPings(pings.select("value"), opts.raiseOnError(),
+      opts.measurementName(), opts.acceptedChannels(), opts.acceptedAppNames())
+      .writeStream
+      .queryName(QueryName)
+      .foreach(httpSink)
+      .start()
+      .awaitTermination()
+  }
+
+  def sendBatchCrashes(spark: SparkSession, opts: Opts): Unit = {
+    val maxParallelRequests = opts.maxParallelRequests()
+
+    implicit val sc = spark.sparkContext
+
+    datesBetween(opts.from(), opts.to.get).foreach { currentDate =>
+      val dataset = com.mozilla.telemetry.heka.Dataset("telemetry")
+      val pings = dataset
+        .where("sourceName") {
+          case "telemetry" => true
+        }.where("docType") {
+          case doctype if doctype == "crash" => true
+        }.where("appUpdateChannel") { // TODO: make channel filtering work
+          case appUpdateChannel if List[String]("beta").contains(appUpdateChannel) => true
```

Comment:
Using `if opts.acceptedChannels.contains(appUpdateChannel)` gives 
```
org.apache.spark.SparkException: Task not serializable
...
Cause: java.io.NotSerializableException: com.mozilla.telemetry.streaming.CrashesToInflux$Opts
```
Any ideas how to deal with this?